### PR TITLE
Kafka secret controller

### LINKED ISSUE #72: Remove Cross-Namespace OwnerReferences & Ensure Resource Deletion
The current implementation is creating the runtime resources (Channel/Dispatcher Service/Deployment) in the **knative-eventing** namespace instead of in the "user" namespace where the KafkaChannel and Knative Subscription exist.  

K8S documentation (below) states that this is not allowed and OwnerReferences should not span namespaces.  Therefore we'd like to remove these controller references and still have guaranteed resource cleanup. Now that we're using the Knative-Eventing controller framework we should be able to achieve this with the available informers and such.

```
Note: Cross-namespace owner references are disallowed by design. This means: 
1) Namespace-scoped dependents can only specify owners in the same namespace, 
and owners that are cluster-scoped. 2) Cluster-scoped dependents can only specify 
cluster-scoped owners, but not namespace-scoped owners.
```

**Note:** It is believed knative-eventing is doing similar things in various implementation (broker/trigger/etc) and we should probably use that as an example.

**Note:** This should focus on the Dispatcher Service/Deployment since the pending refactor of Channel Service/Deployment reconciliation into a separate controller (based on the Kafka Secret) will inherently NOT be cross-namespace and can rely on normal K8S OwnerReferences & Garbage Collection. See Issue #73

### LINKED ISSUE #73: Create Kafka Secret Controller For Channel/Producer Reconciliation
Recently we changed the Channel architecture to consolidate the number of Channel (Producer) Deployments from one-per-KafkaChannel to one-per-kafka-auth-Secret.  We did not however split out the Channel reconciliation into a new controller which is watching the kafka authentication secret (instead of KafkaChannels which now share a Controller/Producer).  

We want to do that as it would be the more correct implementation, and would simplify things a bit as follows...

  - one-to-one relationship instead of many-to-one being managed in controller/reconciler
  - both the secret and Channel resources are all in knative-eventing namespace so we could re-add the OwnerReference and let K8S Garbage Collection handle deletions.

### PR DESCRIPTION
**Pull request by travis-minke-sap**
Separated the reconciliation of the "Channel" Service/Deployment (which are based on the Kafka Secret) into it's own Controller/Reconciler.  

Implemented necessary cross-controller watching in order to update status and refresh the Kafka AdminClient (no longer require restarting controller when new secrets are added).

Split tests into two, added necessary utilities, etc.

Providied additional consistency cleanup.

### CODE REVIEW

File: components/controller/pkg/controller/kafkachannel/kafkachannel.go [hunk 1]
```diff
@@ -1,57 +1,80 @@
...
+	} else {
+		logger.Info("Successfully Reconciled KafkaChannel Labels")
 	}
 
-	// Return Success
-	r.Logger.Info("Successfully Reconciled KafkaChannel", zap.Any("Channel", kafkachannel))
-	kafkachannel.Status.ObservedGeneration = kafkachannel.Generation
-	return reconciler.NewEvent(corev1.EventTypeNormal, event.KafkaChannelReconciled.String(), "KafkaChannel Reconciled Successfully: \"%s/%s\"", kafkachannel.Namespace, kafkachannel.Name)
+	// Return Results
+	if labelsErr != nil {
```
Comment (eric-sap) [inline components/controller/pkg/controller/kafkachannel/kafkachannel.go]: Is there a reason this if statement isn't combined with the one right above it which has an identical condition?  Seems a little odd.

File: components/controller/pkg/controller/kafkachannel/reconciler.go [hunk 2]
```diff
@@ -5,34 +5,86 @@ import (
...
+	kafkachannelLister    knativekafkalisters.KafkaChannelLister
 	kafkachannelInformer  cache.SharedIndexInformer
 	deploymentLister      appsv1listers.DeploymentLister
 	serviceLister         corev1listers.ServiceLister
 }
 
-// Check that our Reconciler implements Interface
-var _ kafkachannelreconciler.Interface = (*Reconciler)(nil)
+// Verify The KafkaChannel Reconciler Implements It's Interface
+//var _ kafkachannelreconciler.Interface = (*Reconciler)(nil)
```
Comment (eric-sap) [inline components/controller/pkg/controller/kafkachannel/reconciler.go]: Why is this here if it is commented out ?

Comment (travis-minke-sap) [reply components/controller/pkg/controller/kafkachannel/reconciler.go]: good catch - leftover from refactor - will remove!

File: components/controller/pkg/controller/kafkachannel/kafkachannel.go (hunk 1 above)
Comment (travis-minke-sap) [reply components/controller/pkg/controller/kafkachannel/kafkachannel.go]: yeah - was leaving the concept of "aggregating" the various reconciliation errors - but I agree it's more confusing this way and easy enough to put back if we need to.  will change

**Review by phamilton — APPROVED** [review]
LGTM
